### ISSUE TITLE
Add ability to specify GCP Compute image storage location

### ISSUE THREAD

**Issue by whiiiskyy** [hashicorp/packer#10789]
Hi guys,

I have been unable to specify the location for the new image using packer.
This same question was asked in [a 2019 post](https://github.com/hashicorp/packer/issues/8109) and I cannot find further information.
The user was running a second script to generate an image from the packer image to be able to set the right zone. 
That is a madness solution.

When packer generates the images, it sets the location to "asia (Asia-Pacific) (default)" which is a multi-regional.
I would like to set it to regional "australia-southeast (Sydney) (default)" but I can't.
I receive an error message:

```
Errors validating build 'googlecompute'. 1 error occurred:
	* unknown configuration key: "image_zone"
```

[Google GCP does accept](https://cloud.google.com/build/docs/building/build-vm-images-with-packer#json) that as you can follow below: 

``` {
  "steps": [
   {
      "name": "gcr.io/[PROJECT_ID]/packer",
      "args": [
        "build",
        "-var",
        "image_name=[IMAGE_NAME]",
        "-var",
        "project_id=[PROJECT_ID]",
        "-var",
        "image_family=[IMAGE_FAMILY]",
        "-var",
        "image_zone=[IMAGE_ZONE]",
        "packer.json"
       ]
    }
   ]
}
```
I am new to packer and trying somehow to get this to work but I am not sure how to implement that into my template.

```
{
  "builders": [{
      "type": "googlecompute",
      "project_id": "ID",
      "source_image": "IMAGE",
      "ssh_username": "USER",
      "machine_type": "TYPE",
      "zone": "australia-southeast1-a",
      "subnetwork": "SUBNET",
      "omit_external_ip": true,
      "use_internal_ip": true,            
      "image_description": "DESCRIPTION",
      "disk_size": 16,
      "image_name": "NAME", 
      "image_family": "FAMILY",
      "image_labels": {"date-created": "{{isotime \"2021-03-19\"}}"},
      "args": [
        "build",
        "-var",
        "image_zone=australia-southeast1-a"
      ]
    }], 
  "provisioners":[{
      "type": "ansible",
      "playbook_file": "tomcat.yml",
      "user": "USER"
    }
  ]}
```
Without the args section above, everything works fine. Build, Ansible, Image, everything is processed.
The problem is when I try to set the image zone.

Thank you

**Comment by nywilken** [comment]
HI there @whiiiskyy thanks for reaching out. Support for setting the storage location came after that original issue you mentioned so that is probably the cause for confusion, as there is no `"args"` field in Packer. There is an `image_storage_locations` field that allows you to change the storage location. The field is an array of one. Here's the link to the field https://www.packer.io/docs/builders/googlecompute#image_storage_locations

Please try replacing `"args"` with `"image_storage_locations": ["australia-southeast1-a"]`  and let us know if it solves the issue you are having. 

Cheers!

**Comment by whiiiskyy** [comment]
@nywilken sorry for the late reply.
I have tried that and I get an error so why I tried something else:

```
packer validate devtest_template.json                                                            ░▒▓ ✔ 
Template validation failed. Errors are shown below.

Errors validating build 'googlecompute'. 1 error occurred:
	* unknown configuration key: "image_storage_locations"
```
I tried both packer 1.3.4 and the latest 1.7 but it doesn't seem to understand that.

Thank you

**Comment by nywilken** [comment]
Hi there @whiiiskyy this particular features was added in v1.6.0 of Packer so it will not work in v1.3.4. As for version v1.7.0 can I see the debug logs `PACKER_LOG=1 packer validate devtest_template.json` and you most recent template, please. I am not able to reproduce this issue using v1.7.0 and my sample template. 

Is there any chance that your v1.7.0 is aliased to an older version of Packer? 


```
{
	"variables" : {
		"zone": "us-central1-a",
		"service_account": "{{env `GOOGLE_APPLICATION_CREDENTIALS`}}",
		"project": "{{env `GOOGLE_PROJECT_ID`}}",
		"ssh_username": "packer"
	},
	"builders" : [
			{
					"type": "googlecompute",
					"image_name" : "packer-tester-{{timestamp}}",
					"project_id": "{{user `project`}}",
					"source_image_family": "centos-8",
					"ssh_username": "{{user `ssh_username`}}",
					"zone": "{{user `zone`}}",
					"image_storage_locations": ["us"]
				}
	],
	"provisioners" : [
			{
					"type" : "shell",
					"inline" : ["ls /var/log"],
					"execute_command" : "sudo -E -S sh '{{ .Path }}'"
			}
 ]
}
```

**Comment by whiiiskyy** [comment]
I don't know what to say haha

I basically did a copy/paste from your example and nothing.
I am running Linux Mint which doesn't see to have an official repository for packer1.7
Either way, I uninstalled 1.3.4 but I was still getting the same error with packer1.7

I change the order within the builders section. 
Everything related to the image afterwards, I moved to the top.
Everything related to the temp VM (internal IP only, network, disk, etc,) were set at the end.

It worked. I will keep this in mind next time.
